src/components/AnimateSharedLayout/utils/stack.ts

```diff
@@ -1,212 +1,144 @@
+        add(element) {
+            element.setCrossfader(crossfader)
+            stack.add(element)
+
+            /**
+             * Hydrate new element with previous drag position if we have one
+             */
+            if (prevDragCursor) element.prevDragCursor = prevDragCursor
+
+            if (!state.lead) state.lead = element
+        },
+        remove(element) {
+            stack.delete(element)
+        },
+        getLead: () => state.lead,
+        updateSnapshot() {
+            if (!state.lead) return
+
+            prevValues = crossfader.isActive()
+                ? crossfader.getLatestValues()
+                : state.lead.getLatestValues()
+            prevViewportBox = state.lead.prevViewportBox
+
+            const dragControls = elementDragControls.get(state.lead)
+            if (dragControls && dragControls.isDragging) {
+                prevDragCursor = dragControls.cursorProgress
             }
-        })
-
-        const dragControls = elementDragControls.get(this.lead)
-        if (dragControls && dragControls.isDragging) {
-            snapshot.isDragging = true
-            snapshot.cursorProgress = dragControls.cursorProgress
-        }
-
-        this.snapshot = snapshot
-    }
-
-    isLeadPresent() {
-        return this.lead && this.lead?.presence !== Presence.Exiting
-    }
-
-    getFollowOrigin(): AxisBox2D | undefined {
-        return this.follow
-            ? this.follow.prevViewportBox
-            : this.snapshot?.boundingBox
-    }
-
-    getFollowTarget(): AxisBox2D | undefined {
-        return this.follow?.box
-    }
+        },
+        clearSnapshot() {
+            prevDragCursor = prevViewportBox = undefined
+        },
+        // TODO We might not need this
+        updateLeadAndFollow() {
+            prevState = { ...state }
+
+            let lead: VisualElement | undefined
+            let follow: VisualElement | undefined
+            const order = Array.from(stack)
+            for (let i = order.length; i--; i >= 0) {
+                const element = order[i]
+                if (lead) follow ??= element
+                lead ??= element
+                if (lead && follow) break
+            }
+            state.lead = lead
+            state.follow = follow
+            state.leadIsExiting = state.lead?.presence === Presence.Exiting
+
+            crossfader.setOptions({
+                lead,
+                follow,
+                prevValues,
+                crossfadeOpacity:
+                    follow?.isPresenceRoot || lead?.isPresenceRoot,
+            })
+
+            if (
+                prevState.lead !== state.lead ||
+                prevState.leadIsExiting !== state.leadIsExiting
+            ) {
+                needsCrossfadeAnimation = true
+            }
+        },
+        animate(child, shouldCrossfade = false) {
+            if (child === state.lead) {
+                if (shouldCrossfade) {
+                    child.pointTo(state.lead)
```

Comment:
Am I right to think that what we're doing here is updating the child to point to itself **in case** it's been pointed to a different element in a previous transition?

If so maybe a comment would be helpful? My initial reaction was confusion as to why an element would need to point to itself 🤔